Framework/DataObjects/inc/MantidDataObjects/MDBox.tcc

```diff
@@ -568,26 +569,32 @@ TMDE(void MDBox)::integrateSphere(Mantid::API::CoordTransform &radiusTransform,
       }
     }
   } else {
-      // For each MDLeanEvent
-    std::vector<double> eventVec, errsqVec;
+    // For each MDLeanEvent
+    using valAndErrorPair = std::pair<signal_t, signal_t>;
+    std::vector<valAndErrorPair> vals;
     for (const auto &it : events) {
       coord_t out[nd];
       radiusTransform.apply(it.getCenter(), out);
       if (out[0] < radiusSquared && out[0] > innerRadiusSquared) {
-        eventVec.push_back(static_cast<signal_t>(it.getSignal()));
-        errsqVec.push_back(static_cast<signal_t>(it.getErrorSquared()));
-        //signal += static_cast<signal_t>(it.getSignal());
-        //errorSquared += static_cast<signal_t>(it.getErrorSquared());
+        valAndErrorPair newPair;
+        newPair.first = static_cast<signal_t>(it.getSignal());
+        newPair.second = static_cast<signal_t>(it.getErrorSquared());
+        vals.emplace_back(std::move(newPair));
```

Comment:
I'm not an expert in this stuff, but wouldn't
```c++
const auto first = static_cast<signal_t>(it.getSignal());
const auto second = static_cast<signal_t>(it.getErrorSquared());
vals.emplace_back(first, second);
```
save you the construction and copying of the temporary `newPair`?